### ISSUE TITLE
[Feature Request] Rename new duck 

### ISSUE THREAD

**Issue by Ogeru** [KitsueFox/PPDS-Mods#3]
Is there any way to automatically (re)name a new duck variant with text we could have in clipboard, or maybe we could send to the mod ?

**Comment by KitsueFox** [comment]
I haven't really messed with renaming ducks but, I'll see what I can do

**Comment by KitsueFox** [comment]
Progress update

![Placid_Plastic_Duck_Simulator_atIkKFWQXE](https://user-images.githubusercontent.com/80623201/221434773-e5cad099-cb40-4dac-9d50-8733dbd93d21.jpg)

**Comment by Ogeru** [comment]
Can't wait to see the result ❤️

**Comment by KitsueFox** [comment]
ok, I've spent some time with this mod idea, it's not automatic yet but the concept is a little close to what you want. The mod currently overrides the in-games naming setup for the ducks but allows better customization for naming the ducks and even change the color and style of the name tag with [Rich Text](https://docs.unity3d.com/Packages/com.unity.ugui@1.0/manual/StyledText.html) as shown down below

![Placid_Plastic_Duck_Simulator_2T1JtRodId](https://user-images.githubusercontent.com/80623201/221664098-8278bb96-da26-464d-89a2-20a516d8ba23.jpg)
It does name the duck on whatever you have in your clipboard, so naming the beeduck the entire "Bee Movie" script is possible. Names do persist along with duplicated ducks

I also have a demo video on what i came up so far
[![CustomName Demo](https://img.youtube.com/vi/kDhJ9IOkwtg/0.jpg)](https://www.youtube.com/watch?v=kDhJ9IOkwtg)

If you have any other ideas feel free to let me know 🙂

PS: You might have to explain your original idea to me

**Comment by Ogeru** [comment]
That's great work so far ! ❤️

It's not really in my current needs but I was also thinking about :
- Are emojis working as name ?
- Could we use like Twitch emote as name ? 
- And what about animated twitch emotes ?

Edit : I'm a french streamer on twitch, working on twitch integrations for some games ;)

**Comment by KitsueFox** [comment]
As far as emojis go, I won't be able to implement that due limitation of the UnityEngine.TextMesh on the Ducks as it doesn't support any emojis

**Comment by Ogeru** [comment]
Hey !

I just tried customnames, but I may have done something wrong. I've put some text in my clipboard with some nodejs; something simple, like "hello world", spawned a new duck, but it has no name on it. 

Is it the intended behavior for the current release of the mod - I mean, not working ?

Btw, not sure if I have to make another issue, but eventreenabler break the duck spawning.

**Comment by KitsueFox** [comment]
I'm still working on it, I just been a bit busy. I'm going to try to finish it this weekend. In the current state the mod is in you basically you copy something into the clipboard, you click on a duck, and them press Numpad 0. I'll try to finish it up during the weekend 

EDIT: PS: I didn't close the issue cause I wasn't done with the mod yet

**Comment by Ogeru** [comment]
Yeah sure, no rush, I just saw the dlls released and had a look 😉

It works great when using numpad 0.

I will patiently wait for updates on the repo !

**Comment by Ogeru** [comment]
Hey ! Any news on the project ?

You may not have the time atm for that, and maybe it's complicated to name the new duck automatically ?

So, I'm trying to figure out how to use the mod as it is right now. I can rename a duck with my clipboard when I select him.

But I can't figure out how to rename the duck that is just spawning.
Is there any way ? If it's complicated or needs work on your side, I'll just wait for the resolution of the issue.

Have a nice day !

**Comment by KitsueFox** [comment]
Sorry for the long wait, life kind of took over but the mod is mostly done minus one bug which is due to Multiplayer

**Comment by Ogeru** [comment]
It's okay, life goes first ! I'll check that tomorrow !

**Comment by Ogeru** [comment]
Hey !
I just checked the update on that issue !
Your work is great but differs from what I expected.
The initial request was to be able to rename the newly spawned duck.

"Tab" allows to select a random duck, but there is no known way to select the just spawned duck to rename it.
Do you have any way to target the new spawned duck in order to give it a name ?

It doesn't bother me to have to press "num0" to rename, but I would like to give provide the name to that new duck.

Thank you !

**Comment by KitsueFox** [comment]
I had forgot to update the Readme.md but while I'm here I'll respond. I have the mod setup for automatically setting the name of the newly spawned Duck based off the clipboard, the feature is disabled by default but you can enable it in the `Placid Plastic Duck Simulator\UserData\MelonPreferences.cfg` and you can set the `AutoName` to true
ex. 
![image](https://github.com/KitsueFox/PPDS-Mods/assets/80623201/b64918de-e2ed-4b77-9990-9e607983ee33)

I'm going to have a mods settings system in the near future when I get some more free time. In the mean time I'll leave the issue open

**Comment by Ogeru** [comment]
That's a great news, thanks for the info ! 

I just tested it, and it looks like there is a bug at a certain point.
The first rename worked great to me, but once the second duck spawns, melonloader console spams with the duck name, and as soon as I copy something else in the clipboard, it renames "duck 1" with the future "duck 2" name in clipboard

And if my clipboard got empty, the duck name disappear too. 
![image](https://github.com/KitsueFox/PPDS-Mods/assets/20920321/4255f5a1-adec-4102-80fa-125aed5222b3)

Please tell me if I can help you debugging that in any way

**Comment by Ogeru** [comment]
Hey there,

Any luck with that bug ?

**Comment by Ogeru** [comment]
Hey there, 

I'm dev but never touched melonloader nor C#.

How Can I help ?
